### PR TITLE
replace the KMD usage in pingpong with direct signing

### LINKED ISSUE #2644: faster pingpong
## Problem

pingpong may currently be limited by the speed of doing API requests to kmd and/or this API connection within machine is inefficient and causes excess CPU load in testing.

## Solution

Improve pingpong to create key pairs internally and do signing within process. ( i.e. instead of using the kmd )

Keep key pairs as msgpack blob in current directory (or location based on flag) so that accounts can be re-used on a subsequent pingpong session.

## Urgency

This will improve testing of the system beyond 1000 TPS

### PR DESCRIPTION
**Pull request by AlgoStephenAkiki**
## Summary

Removes signing calls to KMD by creating internal hashmap and signing transactions locally.

Resolves #2644 

## Test Plan

Tested using performance system.

### CODE REVIEW

**Comment by brianolson** [comment]
Still should have local storage for generated accounts

File: shared/pingpong/pingpong.go [hunk 1]
```diff
@@ -208,12 +220,12 @@ func fundAccounts(accounts map[string]uint64, client libgoal.Client, cfg PpConfi
 	}
 
 	fmt.Printf("adjusting account balance to %d\n", minFund)
-	for addr, balance := range accounts {
+	for addr, val := range accounts {
```
Comment (tsachiherman) [inline shared/pingpong/pingpong.go:223]: `val` -> `acct`

File: shared/pingpong/pingpong.go [hunk 2]
```diff
@@ -39,10 +40,17 @@ type CreatablesInfo struct {
 	OptIns      map[uint64][]string
 }
 
+type pingPongAccount struct {
```
Comment (tsachiherman) [inline shared/pingpong/pingpong.go:43]: add a very short line of comment explaining what this struct is used for.

File: shared/pingpong/pingpong.go [hunk 3]
```diff
@@ -39,10 +40,17 @@ type CreatablesInfo struct {
...
+	sk      *crypto.SignatureSecrets
+	pk      basics.Address
+}
+
 // WorkerState object holds a running pingpong worker
 type WorkerState struct {
-	cfg      PpConfig
-	accounts map[string]uint64
+	cfg PpConfig
+	//accounts map[string]uint64
```
Comment (tsachiherman) [inline shared/pingpong/pingpong.go:52]: drop commented out lines.

File: shared/pingpong/pingpong.go [hunk 4]
```diff
@@ -786,32 +798,25 @@ func (pps *WorkerState) constructTxn(from, to string, fee, amt, aidx uint64, cli
...
-		stxn, err = client.SignTransactionWithWalletAndSigner(h, nil, signer, txn)
+		stxn, err = txn.Sign(accounts[signer].sk), nil
 	} else if len(cfg.Program) > 0 {
 		// If there's a program, sign it and use that in a lsig
-		psig, err = client.SignProgramWithWallet(h, nil, signer, cfg.Program)
-		if err != nil {
-			return
-		}
+
+		psig = accounts[signer].sk.Sign(logic.Program(cfg.Program))
```
Comment (tsachiherman) [inline shared/pingpong/pingpong.go:810]: I could be wrong, but I think that you might want to do
```golang
progb := logic.Program(cfg.Program)
psig = accounts[signer].sk.Sign(&progb)
```

**Review by tsachiherman — CHANGES_REQUESTED** [review]
This PR seems to fail for the following test cases:
asset_transfer_small, asset_transfer_large, stateful_teal_small, stateful_teal_medium, stateful_teal_large.
Could you manually test that locally and make sure it generates valid transactions ?

**Review by tsachiherman — APPROVED** [review]
looks good.

**Pull request merged**